### PR TITLE
tde-mi contracts

### PR DESCRIPTION
**Pull request by junierch**
This PR fixes #
Contracts for tde-mi feature. No UI changes. 
Extracted from reviewed PR https://github.com/microsoft/azuredatastudio/pull/21573/files

+ Adding SQL Tools explicitly because we are modifying mssql code.
@Charles-Gagnon @aasimkhan30

### CODE REVIEW

**Comment by raymondtruong** [comment]
looks good to me, but maybe to be safe we should have some folks from mssql review as well

**Review by raymondtruong — APPROVED** [review]

**Review by neetusingh-ms — APPROVED** [review]

File: extensions/mssql/src/mssql.d.ts [hunk 1]
```diff
@@ -831,4 +833,36 @@ declare module 'mssql' {
 		completedStep: LoginMigrationStep;
 		elapsedTime: string;
 	}
+
+	// SqlMigration interfaces  BEGIN -----------------------------------------------------------------------
```
Comment (aasimkhan30) [inline extensions/mssql/src/mssql.d.ts:837]: NIT: TDEMigration

**Review by aasimkhan30 — APPROVED** [review]
Is there a reason why we are adding a new service and not using existing MigrationService?

File: extensions/mssql/src/mssql.d.ts (hunk 1 above)
Comment (junierch) [reply extensions/mssql/src/mssql.d.ts]: Fixed. Thx.

**Comment by junierch** [comment]
> Is there a reason we are adding a new service and not using the existing MigrationService?

Just exploratory, as an exercise to better understand the services definition structure, we are going to use this service as a concept when building the JSON-RPC backend for the SQL Migration extension.

**Review by kburtram — APPROVED** [review]
I not sure regarding design details of the actual service, but from an integration standpoint with STS it looks consistent with what other services are doing.

**Pull request merged**